TestDNS

```
#177 54.95 === RUN   TestDNS
#177 54.95 === RUN   TestDNS/127.0.0.1:10700/success
#177 54.95     logger.go:146: 2025-05-01T10:15:36.917Z	DEBUG	dns server listener closed	{"net": "udp", "local_addr": "127.0.0.1:10701"}
#177 54.95     logger.go:146: 2025-05-01T10:15:36.927Z	DEBUG	making dns request	{"data": ";; opcode: QUERY, status: NOERROR, id: 9173\n;; flags: rd; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0\n\n;; QUESTION SECTION:\n;google.com.\tIN\t A\n", "proto": "udp", "question": "google.com.", "local_addr": "127.0.0.1:10700", "remote_addr": "127.0.0.1:43935", "upstream": "8.8.8.8:53"}
#177 54.95     logger.go:146: 2025-05-01T10:15:36.931Z	DEBUG	dns response	{"data": ";; opcode: QUERY, status: NOERROR, id: 9173\n;; flags: rd; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0\n\n;; QUESTION SECTION:\n;google.com.\tIN\t A\n", "proto": "udp", "question": "google.com.", "local_addr": "127.0.0.1:10700", "remote_addr": "127.0.0.1:43935", "data": ";; opcode: QUERY, status: NOERROR, id: 9173\n;; flags: qr rd ra; QUERY: 1, ANSWER: 6, AUTHORITY: 0, ADDITIONAL: 0\n\n;; QUESTION SECTION:\n;google.com.\tIN\t A\n\n;; ANSWER SECTION:\ngoogle.com.\t300\tIN\tA\t192.178.218.102\ngoogle.com.\t300\tIN\tA\t192.178.218.139\ngoogle.com.\t300\tIN\tA\t192.178.218.113\ngoogle.com.\t300\tIN\tA\t192.178.218.138\ngoogle.com.\t300\tIN\tA\t192.178.218.100\ngoogle.com.\t300\tIN\tA\t192.178.218.101\n"}
#177 54.95     dns_test.go:100: r: ;; opcode: QUERY, status: NOERROR, id: 9173
#177 54.95         ;; flags: qr rd ra; QUERY: 1, ANSWER: 6, AUTHORITY: 0, ADDITIONAL: 0
#177 54.95         
#177 54.95         ;; QUESTION SECTION:
#177 54.95         ;google.com.	IN	 A
#177 54.95         
#177 54.95         ;; ANSWER SECTION:
#177 54.95         google.com.	300	IN	A	192.178.218.102
#177 54.95         google.com.	300	IN	A	192.178.218.139
#177 54.95         google.com.	300	IN	A	192.178.218.113
#177 54.95         google.com.	300	IN	A	192.178.218.138
#177 54.95         google.com.	300	IN	A	192.178.218.100
#177 54.95         google.com.	300	IN	A	192.178.218.101
#177 54.95     logger.go:146: 2025-05-01T10:15:36.931Z	DEBUG	dns server listener closed	{"net": "tcp", "local_addr": "127.0.0.1:10700"}
#177 54.95     logger.go:146: 2025-05-01T10:15:36.931Z	DEBUG	dns server listener closed	{"net": "udp", "local_addr": "127.0.0.1:10700"}
#177 54.95 === RUN   TestDNS/127.0.0.1:10700/failure
#177 54.95     logger.go:146: 2025-05-01T10:15:36.933Z	DEBUG	dns server listener closed	{"net": "udp", "local_addr": "127.0.0.1:10701"}
#177 54.95     logger.go:146: 2025-05-01T10:15:36.943Z	DEBUG	making dns request	{"data": ";; opcode: QUERY, status: NOERROR, id: 43211\n;; flags: rd; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0\n\n;; QUESTION SECTION:\n;google.com.\tIN\t A\n", "proto": "udp", "question": "google.com.", "local_addr": "127.0.0.1:10700", "remote_addr": "127.0.0.1:51167", "upstream": "242.242.242.242:53"}
#177 54.95     logger.go:146: 2025-05-01T10:15:38.944Z	WARN	error serving dns request	{"error": "read udp 192.168.72.134:33775->242.242.242.242:53: i/o timeout"}
#177 54.95     check.go:35: 
#177 54.95         	Error Trace:	/.cache/mod/github.com/siderolabs/gen@v0.8.0/xtesting/check/check.go:35
#177 54.95         	            				/src/internal/pkg/dns/dns_test.go:94
#177 54.95         	Error:      	An error is expected but got nil.
#177 54.95         	Test:       	TestDNS/127.0.0.1:10700/failure
#177 54.95     logger.go:146: 2025-05-01T10:15:38.944Z	DEBUG	dns server listener closed	{"net": "udp", "local_addr": "127.0.0.1:10700"}
#177 54.95     logger.go:146: 2025-05-01T10:15:38.944Z	DEBUG	dns server listener closed	{"net": "tcp", "local_addr": "127.0.0.1:10700"}
#177 54.95 --- FAIL: TestDNS (2.03s)
#177 54.95     --- PASS: TestDNS/127.0.0.1:10700/success (0.02s)
#177 54.95     --- FAIL: TestDNS/127.0.0.1:10700/failure (2.01s)
#177 54.95 FAIL
#177 54.95 coverage: 0.3% of statements in github.com/siderolabs/talos/...
#177 54.95 FAIL	github.com/siderolabs/talos/internal/pkg/dns	2.168s
```

--- a/internal/pkg/dns/dns_test.go
+++ b/internal/pkg/dns/dns_test.go
@@ -90,7 +90,11 @@ func TestDNS(t *testing.T) {
 
 				time.Sleep(10 * time.Millisecond)
 
-				r, err := dnssrv.Exchange(createQuery(test.hostname), dnsAddr)
+				c := dnssrv.Client{
+					Timeout: time.Second,
+				}
+
+				r, _, err := c.Exchange(createQuery(test.hostname), dnsAddr)
 				test.errCheck(t, err)
 
 				if r != nil {

